lemur/common/celery.py

```diff
@@ -149,6 +149,15 @@ def report_celery_last_success_metrics():
         metrics.send(
             f"{task}.time_since_last_success", "gauge", current_time - last_success
         )
+
+        kwargs = t.get("kwargs")
+        if "source" in kwargs:
+            source = kwargs["source"]
+            last_success = int(red.get(f"{source}.last_success") or 0)
+            metrics.send(
+                f"{source}.time_since_last_success", "gauge", current_time - last_success
```

Comment:
What would be your thoughts be on storing the value of `source` in a metric tag so we can potentially write a single alert rule that covers all source rotations?

On a similar note, do you think there's anything we can do here to gracefully handle source deletions? E.g. avoid emitting a metric about it forever, or triggering false positive alerts.

Reply:
Adding the Source to a metric tag is a good idea, I didn't think of that. So we could potentially have something like
```python
metrics.send(
  f"{task}.time_since_last_success", 
  "gauge",
   current_time - last_success, 
  tags={"source_name": source}
)
```

Reply:
Will the use of tags inherently solve the "no data" issue? When we delete a source we will stop sending the metric for that specific source but we will continue sending the metric for the other sources.

Reply:
We may still get a "no data" alert for the series that metrics were stopped being reported for. However, we could configure our alerts to automatically resolve from a no data state after XY hours. It's not perfect, but this will avoid needing to create and maintain 20+ alert rule definitions to work around a rare edge case where a source is removed. Plus, it seems if we created 20+ alert rule definitions, we'd run into the same problem anyway.